src/content/blog/fastify-2.mdx

```diff
@@ -0,0 +1,98 @@
+---
+title: "Moving to fastify pt. 2: Docs, an SDK and AI tools"
+description: |
+  Taking advantage of our typed REST API to build a platform around
+  Val Town.
+pubDate: July 12, 2024
+author: Tom MacWright
+---
+
+The last time, I [wrote about how we were switching from express to fastify][last_post]
+for Val Town’s REST API. To recap, Fastify gives us a way to strictly
+define our routes, with all of the path and query string parameters,
+request and response bodies aligning to a specific schema. This is a big
+win for robustness, gives us great type-safety with TypeScript, and lets us
+generate an [OpenAPI Specification][openapi] directly from our server code.
+
+But typesafety and robustness don't bring home the bacon. What do you,
+the user, get from our time spent refactoring?
+
+### Nice new REST API documentation
+
+![Val Town REST API Documentation](./fastify-2/docs.png)
+
+First thing, we get **way better REST API documentation**. It has example
+requests, with everything from cURL to Python clients. It describes the precise
+shapes of the objects you'll get back. You can even test APIs right in
+the API docs. Hats off to [Scalar](https://scalar.com), whose
+[open source package](https://github.com/scalar/scalar) is powering this.
+
+The docs are based on our OpenAPI spec, which is emitted from Fastify and
+pulled from our server's routes. From there on, setting up nice documentation
+is basically as simple as this:
+
+```html
+<script id="api-reference" data-url="./openapi.documented.json"></script>
+<script src="https://cdn.jsdelivr.net/npm/@scalar/api-reference"></script>
+```
+
+### A TypeScript SDK
+
+Are you not entertained? How about [a new TypeScript SDK](https://github.com/val-town/sdk)?
+
+```ts title="example.ts"
+// Import the SDK
+import ValTown from "@valtown/sdk";
+// Initialize it
+const valTown = new ValTown();
+// Use it
+console.log(await valTown.me.profile.retrieve());
+```
+
+The SDK covers all of our major API methods, and has robust TypeScript
+types for everything. Methods in our API that are paginated, like listing
+comments, vals, or search results, are implemented as async iterators
+in the SDK so you don't even have to implement any pagination logic.
+
+Just install it.
```

Comment:
I think we should prioritize showing off its use in Val Town first and foremost (link to a val or two?) and then say of course it works in all the other js runtimes